packages/dds/tree/src/test/memory/tree.spec.ts

```diff
@@ -33,28 +39,65 @@ class RootNodeSchema extends builder.object("root-item", {
 	child: builder.optional(ChildNodeSchema),
 }) {}
 
-function createLocalSharedTree(id: string): TreeView<typeof RootNodeSchema> {
-	const sharedTree = SharedTree.create(
+const initialState = {
+	child: {
+		propertyOne: 128,
+		propertyTwo: {
+			itemOne: "",
+		},
+		propertyThree: new Map([["numberOne", 1]]),
+	},
+};
+
+// Array has 2 allowable types to make it less efficient for uniform chunks.
+class RootNodeSchemaBasicChunks extends builder.array("root-item-with-basic-chunks", [
+	builder.number,
+	builder.string,
+]) {}
+
+class RootNodeSchemaUniform extends builder.array(
+	"root-item-with-basic-chunks",
+	builder.number,
+) {}
+
+class NestedNodeSchema extends builder.object("wrapped-item", {
+	layer1: builder.object("nested-1", {
+		layer2: builder.object("nested-2", {
+			layer3: builder.object("nested-3", {
+				layer4: builder.object("nested-4", {
+					x: builder.number,
+					y: builder.number,
+				}),
+			}),
+		}),
+	}),
+}) {}
+
+// Array with nodes which are nested
+class RootNodeSchemaWithNestedNodes extends builder.array(
+	"root-item-with-nested-nodes",
+	NestedNodeSchema,
+) {}
+
+function createLocalSharedTree<TSchema extends ImplicitFieldSchema>(
+	id: string,
+	schema: TSchema,
+	content: InsertableTreeFieldFromImplicitField<TSchema>,
+	sharedTreeOptions?: SharedTreeOptions,
+): TreeView<TSchema> {
+	const sharedTree =
+		sharedTreeOptions !== undefined ? configuredSharedTree(sharedTreeOptions) : SharedTree;
```

Comment:
I think this is cleaner since it avoids special casing no options:


```suggestion
	sharedTreeOptions: SharedTreeOptions = {},
): TreeView<TSchema> {
	const sharedTree = configuredSharedTree(sharedTreeOptions);
```